platforms/core-configuration/base-diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/ProjectDetails.java

```diff
@@ -59,14 +68,23 @@ public String getDisplayName() {
 
         @Nullable
         @Override
-
         public String getDescription() {
             return description;
         }
 
+        @Override
+        public String getRelativeProjectDir() {
+            return relativeProjectDirPath;
+        }
+
+        @Override
+        public String getAbsoluteProjectDir() {
+            return absoluteProjectDirPath;
+        }
+
         @Override
         public int hashCode() {
-            return Objects.hash(displayName, description);
+            return Objects.hash(displayName, description, relativeProjectDirPath, projectLogicalPath);
```

Comment:
Absolution path left out as a minor optimization, as it should always track with relative path.